core/include/moveit/task_constructor/storage.h

```diff
@@ -100,6 +100,9 @@ class InterfaceState
 	InterfaceState(const planning_scene::PlanningScenePtr& ps);
 	InterfaceState(const planning_scene::PlanningSceneConstPtr& ps);
 
+	// provide an initial priority for the state
+	InterfaceState(const planning_scene::PlanningSceneConstPtr& ps, const Priority& p);
```

Comment:
I'm not sure we should publicly expose such an interface. There is no reason a user will ever create an interface with an initialized priority. If you need this for testing, provide a utility class there?

Reply:
The priority property is actually private, so a utility class does not help.
If it makes you feel any better, I could also change the visibility to protected instead and inherit just to set an initial priority.

It feels rather absurd to me to have a data structure around that cannot *at all* initialize one of its members though (priority can only be set by `SolutionBase` and updated by `Interface` via `friend` definitions).

Eventually, I only need this to get around [a somewhat questionable assert](https://github.com/ros-planning/moveit_task_constructor/blob/master/core/src/storage.cpp#L106). If the `InterfaceState` does not have any `SolutionBase` objects connected *it can't* be initialized with the current code.
I would be just as happy to remove the assert. ;)

Reply:
> It feels rather absurd to me to have a data structure around that cannot at all initialize one of its members though (priority can only be set by SolutionBase and updated by Interface via friend definitions).

I just want to avoid the risk of users messing around with the priority, e.g. wrongly initializing it...

> Eventually, I only need this to get around a somewhat questionable assert.

Maybe the assert is wrong. Do you have example code that violates the assert?

> If the InterfaceState does not have any SolutionBase objects connected it can't be initialized with the current code.

The default constructor should initialize the priority to `(0, 0.0)`.

Reply:
> Maybe the assert is wrong.

It's not *wrong* in itself, but inconvenient. It is unreachable without the private interfaces.
But using the private interfaces, it is still not possible to fulfill it with a custom `InterfaceState` without a full Solution attached.
It's currently not possible to add a standalone `InterfaceState` to an `Interface` (e.g. via
`InterfaceState is{...}; state->pimpl()->starts().add(is);`

If you deem it better, I can also remove the assert (and thus allow new `InterfaceState` entries with zero priority).
That's still very restrictive because it does not allow to sort multiple new entries by priority, but it fulfills my need.